fossas/fossa-cli
# [ANE-1092] Path Dependency Support

# Overview

This PR is part of the chain, and will not be merged until `CORE` pr is merged!

- modifies `go list` tactic, such that `path+` dependencies are supported!
- adds support for `path+` dependencies 

Sibling PR: https://github.com/fossas/FOSSA/pull/11311

I will put up second PR, once CORE PR is approved, to avoid waste work. Second PR will include docs, and minor optimization to avoid rescanning, scanned dependency, as well as better handling for `checkForEmptyUpload` 

## Acceptance criteria

As a user, 
- [ ] I can successfully scan go project, which has vendor replace directives
- [ ] I can successfully scan go project, which has vendor replace directives
- [ ] `fossa-cli` does not license scan path dependencies, if path dependencies are not supported by endpoint
- [ ] `fossa-cli` does not license scan path dependencies, if output mode is used (`-o` flag)
- [ ] `fossa-cli` can license scan path dependencies, for matching data
- [ ] `fossa-cli` can license scan path dependencies, for fullfile data

## Testing plan

- I relied on automated testing.
- Testing plan will be included in second/final pr of the chain.

However, if you wish you can load the endpoint from: https://github.com/fossas/FOSSA/pull/11311

1. Create example: (refer to repro case here: https://fossa.atlassian.net/browse/ANE-383)
2. Build `make install-dev`
3. Perform `fossa-dev analyze -e http://localhost:9578 --fossa-api-key <KEY> repro`
4. You should see path dependencies on the endpoint

## Risks

N/A

## References

https://fossa.atlassian.net/browse/ANE-1092

## Checklist
